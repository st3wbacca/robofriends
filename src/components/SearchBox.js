import React from 'react';

const SearchBox = ({ searchChange }) => {
	console.log('SearchBox');
	return (
		<div className='pa2'>
			<input
				aria-label='Search Robots'
				className='pa3 ba b--green bg-lightest-blue'
				type='search'
				placeholder='searchrobots'
				onChange={searchChange}
			/>
		</div>
	);
};

export default SearchBox;