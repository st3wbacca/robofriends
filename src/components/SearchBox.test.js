import { shallow } from 'enzyme';
import React from 'react';
import '../setupTests';
import SearchBox from './SearchBox';

it('expect SearchBox to match Snapshop', () => {
	expect.assertions(1);

	const wrapper = shallow(<SearchBox />);
	expect(wrapper).toMatchSnapshot();
});