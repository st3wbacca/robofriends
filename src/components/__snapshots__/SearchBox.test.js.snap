// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect SearchBox to match Snapshop 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBox />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        aria-label="Search Robots"
        className="pa3 ba b--green bg-lightest-blue"
        placeholder="searchrobots"
        type="search"
      />,
      "className": "pa2",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-label": "Search Robots",
        "className": "pa3 ba b--green bg-lightest-blue",
        "onChange": undefined,
        "placeholder": "searchrobots",
        "type": "search",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          aria-label="Search Robots"
          className="pa3 ba b--green bg-lightest-blue"
          placeholder="searchrobots"
          type="search"
        />,
        "className": "pa2",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-label": "Search Robots",
          "className": "pa3 ba b--green bg-lightest-blue",
          "onChange": undefined,
          "placeholder": "searchrobots",
          "type": "search",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
